src/reducers/updateThoughts.ts

```diff
@@ -62,6 +62,15 @@ const updateThoughts = (state: State, { thoughtIndexUpdates, contextIndexUpdates
 
   const recentlyEditedNew = recentlyEdited || state.recentlyEdited
 
+  //  lexemes from the updates that are not available in the state yet.
+  const pendingLexemes = Object.keys(thoughtIndexUpdates).reduce<Index<string>>((acc, thoughtId) => {
+    const lexemeInState = state.thoughts.thoughtIndex[thoughtId]
+    return {
+      ...acc,
+      ...lexemeInState ? {} : { [thoughtId]: thoughtId }
```

Comment:
Why use `{ [thoughtId]: thoughtId }` instead of an array of `thoughtId`? Do you need O(1) lookup? 

If so, I'd suggest `{ [thoughtId]: true }` anyway which is a little cleaner and forces downstream code to use `Object.keys` instead of either `Object.keys` and `Object.values`, which could result in mixed use.

Reply:
> Why use `{ [thoughtId]: thoughtId }` instead of an array of `thoughtId`? Do you need O(1) lookup?

Yes. We need O(1) lookup here in `getSyncedThoughtsUpdate`. Else we will have to iterate the array every time to check if a particular lexeme was pending or not.

https://github.com/shresthabijay/em/blob/5ec0459fe264dd8ab5f8683fac5a2976cd12e2db/src/redux-middleware/pushQueue.ts#L144